extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java

```diff
@@ -44,47 +48,48 @@ public class HttpAuthenticator {
     private final HttpAuthenticationMechanism[] mechanisms;
     private final SecurityEventHelper<AuthenticationSuccessEvent, AuthenticationFailureEvent> securityEventHelper;
 
-    public HttpAuthenticator(IdentityProviderManager identityProviderManager,
-            Event<AuthenticationFailureEvent> authFailureEvent,
-            Event<AuthenticationSuccessEvent> authSuccessEvent,
-            BeanManager beanManager,
-            Instance<HttpAuthenticationMechanism> httpAuthenticationMechanism,
+    HttpAuthenticator(IdentityProviderManager identityProviderManager,
             Instance<IdentityProvider<?>> providers,
+            InjectableInstance<BasicAuthenticationMechanism> basicAuthMechanismInstance,
+            Event<AuthenticationFailureEvent> authFailureEvent, Event<AuthenticationSuccessEvent> authSuccessEvent,
+            BeanManager beanManager, @All List<HttpAuthenticationMechanism> httpAuthenticationMechanisms,
             @ConfigProperty(name = "quarkus.security.events.enabled") boolean securityEventsEnabled) {
         this.securityEventHelper = new SecurityEventHelper<>(authSuccessEvent, authFailureEvent, AUTHENTICATION_SUCCESS,
                 AUTHENTICATION_FAILURE, beanManager, securityEventsEnabled);
         this.identityProviderManager = identityProviderManager;
-        List<HttpAuthenticationMechanism> mechanisms = new ArrayList<>();
```

Comment:
IMHO it is not correct, if the mechanism went to the trouble of implementing the credential types then indeed, if no matching provider is available it is a problem.

Reply:
If the mechanism didn't implement credentials. we always add it. So what is the difference to the previous state if this is not correct? I'm not sure what is a goal then.

Reply:
@michalvavrik Sure, let me close this one and continue in another comment